gpcontrib/gp_debug_numsegments/gp_debug_numsegments--1.0.sql

```diff
@@ -36,3 +41,7 @@ LANGUAGE C IMMUTABLE STRICT EXECUTE ON COORDINATOR;
 CREATE OR REPLACE FUNCTION gp_debug_get_create_table_default_numsegments() RETURNS text
 AS '$libdir/gp_debug_numsegments','gp_debug_get_create_table_default_numsegments'
 LANGUAGE C IMMUTABLE STRICT EXECUTE ON COORDINATOR;
+
+PERFORM pg_catalog.set_config('search_path', old_path, true);
+END
+$$;
```

Comment:
Do  we must use `plpgsql` extension here? Could a simple `reset search_path` or sth like that work well?

Reply:
[`RESET`](https://www.postgresql.org/docs/current/sql-reset.html) set the `search_path` to the default value not the prior value, so it may cannot solve the issue. Ways to creating a variable in PG [including plpgsql and CTE](https://stackoverflow.com/questions/1490942/how-to-declare-a-variable-in-a-postgresql-query), I suppose using plpgsql is the only sensible option here. And it's also **heavily** used by PG extensions already. Many PG extensions are using plpgsql **just** to preserve user `search_path`.

Reply:
OK.